pkg/api/openshiftcluster.go

```diff
@@ -808,4 +808,5 @@ type Identity struct {
 	Type                   string                 `json:"type,omitempty"`
 	UserAssignedIdentities UserAssignedIdentities `json:"userAssignedIdentities,omitempty"`
 	IdentityURL            string                 `json:"identityURL,omitempty" mutable:"true"`
+	TenantID               string                 `json:"tenantId,omitempty" mutable:"true"`
```

Comment:
Should this be a pointer field?
```suggestion
	TenantID               *string                 `json:"tenantId,omitempty" mutable:"true"`
```

Reply:
It's a regular string field in the AKS RP, so I think we should do the same thing as them to make sure we don't run into issues with the Swagger API spec validation later on.

Reply:
I say we should keep this a string. Reasoning:
1. Identity URL is a string, and TenantID is just another MSI header we get from ARM
2. Pointers increase the likelihood of panics via nil pointers.